api/tests/routes/pro/patch_collective_offer_template_test.py

```diff
@@ -200,6 +214,38 @@ def test_unknown_national_program(self, pro_client, offer):
         assert response.status_code == 400
         assert response.json == {"global": ["National program not found"]}
 
+    def test_contact_form_all_fields_null(self, pro_client, offer, payload):
+        payload["contactEmail"] = None
+        payload["contactPhone"] = None
+        payload["contactUrl"] = None
+        payload["contactForm"] = None
+
+        with patch(PATCH_CAN_CREATE_OFFER_PATH):
+            response = pro_client.patch(f"/collective/offers-template/{offer.id}", json=payload)
+
+        assert response.status_code == 400
+        assert "contact[all]" in response.json
+
+    def test_contact_form_both_form_and_url_null(self, pro_client, offer, payload):
+        payload["contactUrl"] = None
+        payload["contactForm"] = None
+
+        with patch(PATCH_CAN_CREATE_OFFER_PATH):
+            response = pro_client.patch(f"/collective/offers-template/{offer.id}", json=payload)
+
+        assert response.status_code == 400
+        assert "contact[url,form]" in response.json
+
+    def test_contact_form_both_form_and_url_set(self, pro_client, offer, payload):
+        payload["contactUrl"] = "https://example.com/contact"
+        payload["contactForm"] = educational_models.OfferContactFormEnum.FORM.value
+
+        with patch(PATCH_CAN_CREATE_OFFER_PATH):
+            response = pro_client.patch(f"/collective/offers-template/{offer.id}", json=payload)
+
+        assert response.status_code == 400
+        assert response.json == {"__root__": ["can't set url with email and/or phone, form"]}
```

Comment:
En phase, mais le message devrait être `Url and form are both not null` ou quelque chose du genre